### ISSUE TITLE
PyLance automatically removes tab when using colon

### ISSUE THREAD

**Issue by Momodips** [microsoft/pylance-release#4507]
## Environment

- Windows 10 Version 22H2
- Visual Studio Code Version 1.78.2 (installed over official website, last stable Windows version)
- Pylance v2023.5.30

## Bug

Every time I use a colon in an `elif`-statement in Python, PyLance erases one tab.

For example:
```
a ="fdiofdjiof"

if True:
    if a[10:] == "":
        pass
elif a[10:] == "":
```
That `elif` statement should have one tab more, but when I pressed 'Shift' + '.' (a colon on a QWERTZ keyboard), it automatically erased one tab. I think the bug only appears when using a colon in a string array selection. Normally it should look like that:

![2mjTk](https://github.com/microsoft/pylance-release/assets/134764811/9c410f93-888b-4e5d-bc91-bb418b4c2d5c)

I downgraded PyLance to v2022.2.4 and that solved the problem.

Here's my Stackoverflow post for that problem:
https://stackoverflow.com/questions/76309201/visual-studio-code-automatically-deletes-one-tab-when-using-a-colon-in-an-elif

**Comment by debonte** [comment]
@Momodips, thanks for reporting this.

If this is causing you a lot of frustration, you could disable the auto-indent feature while you're waiting for this to be fixed by setting:

```json
"[python]": {
  "editor.formatOnType": false
},
```

**Comment by debonte** [comment]
```python
if True:
    if a[10:] == "":
        x = 1
    elif a[10:] == "":
```

**Repro steps:**
1. Delete the `:` within `a[10:]` on the last line.
2. Re-add the colon

**Actual:** Modified line is dedented.

**Expected:** No indentation change since the `elif` is already matched to an `if` at the same indentation level. If the user want to dedent it to match the `if True:` instead they'll need to do it manually.

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2023.7.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023721-12-July-2023-PreRelease)
